Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/orient_polygon_soup_extension.h

```diff
@@ -0,0 +1,194 @@
+// Copyright (c) 2019 GeometryFactory (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-Commercial
+//
+//
+// Author(s)     : Sebastien Loriot and Maxime Gimeno
+
+#ifndef CGAL_ORIENT_POLYGON_SOUP_EXTENSION_H
+#define CGAL_ORIENT_POLYGON_SOUP_EXTENSION_H
+
+#include <CGAL/license/Polygon_mesh_processing/repair.h>
+
+#include <CGAL/AABB_tree.h>
+#include <CGAL/AABB_face_graph_triangle_primitive.h>
+#include <CGAL/AABB_traits.h>
+
+#include <CGAL/Polygon_mesh_processing/shape_predicates.h>
+#include <CGAL/Polygon_mesh_processing/compute_normal.h>
+#include <CGAL/Polygon_mesh_processing/orient_polygon_soup.h>
+
+#include <boost/iterator/counting_iterator.hpp>
+#include <boost/iterator/filter_iterator.hpp>
+
+#ifdef CGAL_LINKED_WITH_TBB
+#include <tbb/parallel_for.h>
+#endif // CGAL_LINKED_WITH_TBB
+
+
+namespace CGAL {
+
+namespace Polygon_mesh_processing {
+
+/*!
+ * \ingroup PMP_orientation_grp
+ * Duplicates each point \a p at which the intersection
+ * of an infinitesimally small ball centered at \a p
+ * with the polygons incident to it is not a topological disk.
+ *
+ * @tparam PointRange a model of the concepts `RandomAccessContainer`
+ * and `BackInsertionSequence` whose value type is the point type.
+ * @tparam PolygonRange a model of the concept `RandomAccessContainer`
+ * whose value_type is a model of the concept `RandomAccessContainer`
+ * whose value_type is `std::size_t`.
+ *
+ * @param points points of the soup of polygons. Some additional points might be pushed back to resolve
+ *               non-manifoldness or non-orientability issues.
+ * @param polygons each element in the vector describes a polygon using the index of the points in `points`.
+ *                 If needed the order of the indices of a polygon might be reversed.
```

Comment:
The doc above says that `PolygonRange` is model of `RandomAccessContainer` but does not specify that is should be modifiable (idem for its value type). I guess reversing the order of a polygon modifies the polygon range though.
Is there a "model of **" missing?